src/math.rs

```diff
@@ -8,3 +8,45 @@ pub(crate) fn clamp(a: i32) -> i32 {
         a
     }
 }
+
+#[inline]
+fn fract(a: f64) -> f64 {
+    if a == 0.0 {
+        0.
+    } else {
+        a % 1.
+    }
+}
+
+#[inline]
+pub(crate) fn round(a: f64) -> f64 {
```

Comment:
```suggestion
pub(crate) fn round(a: f64) -> f64 {
    (if x < 0. { x - 0.5 } else { x + 0.5 }) as i64 as f64
```
For std context it is better to use f64::round. Many CPU has built-in roundings.

Reply:
Since `round` and `clamp` are always called together in the same way, I've just merged the two functions together. Doing so [largely nullifies](https://godbolt.org/#z:OYLghAFBqd5TKALEBjA9gWwA4EsA2ApgE4A0uIEGOBJIxAjABwBsADKQGYEAuJAzpQBGIAORtRpIbgB2AQ2IBPMQ0nT5SgPJCAVoVQ8Vk6pkIyemmfmXjJAE0JCArsADCc/PkGjVpB5jkZYCIxCT9cYn0eXAA3Qm8wwgAPfSc%2BI3JzQnxQyXxcIWIFRVd0B1zSHmJcTAy44iF0fkIAEUIAoPyg3IBKLnw5YH4ANVxCAHdNbDMMznRzAGVUD0IQBgAWLnmeAFVmgAUkip010gHu4id%2BHjOCwUg%2B9Gxo%2Be8AWlcAaieeN6I4/AAUgAzC1gZJ0PVqg57hA%2Bs0iAZcPNKGY7KV8E5MDJTmiADKyQgAOSxQjoDFI2CauBeMgxWJxFKp/BpyJkBJkxNJ5NICKibIWPAUPHp2NOfKR80Fwo5XMwZOI4qFxBF6ExYop1xlhJJ8vJ8PQTmIqFWDD6OVEAFZJDjbKR0GJPpdrqhPgwAHSsd1sT4QACidhp6GIn0BACZgWbJIZbD0%2BgBrECWsPuhgATmBlvWAHY2KxgUxLWmw2cxOtJLVLSx3eta3X6/XSLawg7RJJBBwYxI%2BnBYCgMA4A0GyBQINwiPJTLNtksVmtNnNzHtCIdjqdzqBnTcJbTUTJ0WqGbj97LdQqQCXmaz5qLGZTqbTT9zFSWdwLlar1Xe31KP0%2B9S%2BvIfreSrapyZ50GGBpGiaYjhsCYBgICloAEIyOgAD61x2MhLSAmwACC%2BFERGyEobIXSEBAnL1D0uHEdgThCJ8GAyNcnycDILEDDgGHEIa%2B4QHIIAcSw6w9J8bwgn6ny4MCYahtmKHEZ8qmyZwnxyKGwJfGw7qKcphFqcZnxsCpamAtmLSfNkzTqZp2kyWGlqWvplmGQRJlqc5lrmaplnWbZhAGX5JlCaGYYoaZ7qWhJcj8LJ8mhQFxEBaI5piNaTYVK2ABKVw8J8/CGsawXwb4XbpX0SCEHIDiKnCpaiOWpC1OswIxS5XXdV1LDZXarbtiAna9AmSYpummY5nmLAFkWJYWuC/UtmI0a9KQvYIPAED9lgeBECOlDUPtdCMKwHDjnwxCCBAIh2uoxQZA9Wi6FEGQmGYFhWDYYQOM4bgeF4GT%2BIEwSrHagaREicQJJIySpOkPiSLIfAWmE%2BSFMUpTlHaVQ1HUJCNM0bQdMEsigLG/SDCMYyTNMtq%2BIuPCziEGxbEuBxHHaJwLaD9AFbcQiwo8zxsu8Xw/H8hAAiCYIQlCuAwttk6rIQga0qQMQeE44NhmwYaqHGvLZPyKIQGiIEUviOrPqcV60pb94so%2BNsAeKJuSjI0oqo7P5e3%2BrvnpqwGHhqQFgXKQfQaVF4ZVaNo5Y6W6uh6Xo%2Bv66vBhF8k9GtsZjTmnpsGJyZpgwwIsBsbBppaTUtZWHDNhCq28sNlTrZtO1oGUhBDjwwbkJQ45mHIU5I%2BzzPLKzC7bMuq7cxeZx81uxuIru5v7o71vgbbmz22yjv7/M/7npsfve5%2BR6ah7tIXyfdBnyHX6gSq9%2BKuJvIlbBojwYhZHoVhHgOFLR4UIsRZIV0uKoCKHwIqQCQQeVSqRVCFFCTUWliQOiICGJMQ4lAni2A%2BICTsEJESnAxISSksCGSckFLuVCrgDSWkQS6TckpUKxkzJGQslZGyXhgqMIctJT4Pk2EeS8qpHyyVeFBRCtwry5D1ggHoMQoScUEq0OkaAoiVliLpSallJu9oxD5XYsVGCZUIxhjzt2aqtV6qUDjvXJMjdE5tlbiNfOpBEyFyYMXdYpdy6V3WNXWui0E4DRbpVI2FprHLWbu46JfR6gshROsIAA) any advantage native `f64::round` has.

Reply:
This is true for x86 only. As I mentioned [many CPU](https://godbolt.org/z/7Eb5voj8e) can do this in one instruction.

Reply:
While that is a difference, it's only 2 assembly instructions. I would defer this level of optimisation to a follow-up PR. I'd really rather avoid adding a whole `std` feature gate for a relatively minor performance gain...